profiler/src/ProfilerEngine/Datadog.Profiler.Native.Linux/CrashReportingLinux.cpp

```diff
@@ -0,0 +1,271 @@
+#include "CrashReportingLinux.h"
+
+#include <algorithm>
+#include <cstdint>
+#include <vector>
+#include <dirent.h>
+#include <string>
+#include <memory>
+
+#include <libunwind.h>
+#include <libunwind-ptrace.h>
+#include <sys/ptrace.h>
+#include <sys/wait.h>
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <map>
+#include <string.h>
+#include "FfiHelper.h"
+
+extern "C"
+{
+#include "datadog/common.h"
+#include "datadog/profiling.h"
+}
+
+CrashReporting* CrashReporting::Create(int32_t pid)
+{
+    auto crashReporting = new CrashReportingLinux(pid);
+    return (CrashReporting*)crashReporting;
+}
+
+CrashReportingLinux::CrashReportingLinux(int32_t pid)
+    : CrashReporting(pid)
+{
+}
+
+CrashReportingLinux::~CrashReportingLinux()
+{
+    if (_addressSpace != nullptr)
+    {
+        unw_destroy_addr_space(_addressSpace);
+    }
+}
+
+int32_t CrashReportingLinux::Initialize()
+{
+    auto result = CrashReporting::Initialize();
+
+    if (result == 0)
+    {
+        _addressSpace = unw_create_addr_space(&_UPT_accessors, 0);
+
+        if (_addressSpace == nullptr)
+        {
+            return 999;
```

Comment:
what is the meaning of `999` ? is used somewhere else ?

Reply:
No particular meaning, it's just a non-zero return code